src/compiler/checker.ts

```diff
@@ -33444,8 +33444,9 @@ namespace ts {
                 if (iterationTypes === noIterationTypes) {
                     if (errorNode) {
                         reportTypeNotIterableError(errorNode, type, !!(use & IterationUse.AllowsAsyncIterablesFlag));
-                        errorNode = undefined;

```

Comment:
The reason I kept looping was to provide some kind of useful type for iteration once they fixed the error. The problem the issue mentioned was that no error was reported.

Reply:
(btw: I'm not saying this change is wrong, just expressing the reasoning behind the existing logic).

Reply:
Oh, that makes sense in theory, but the problem here was that `errorNode` was intentionally not supplied, so the non-iterableness of the type was completely lost.

Reply:
What do you think about the baseline change?

Reply:
It seems fine. I do wonder about the `| null` case. If it were `data[Symbol.iterator]()` we would have errored on `data` with `Object is possibly 'null'.`.